### DISCUSSION TITLE
DEV SESSION 14 Oct --- 2000 UTC+8/1400 CET/1200 GMT

### DISCUSSION THREAD

**Discussion by tpike3** [mesa/mesa#1822] [General]
Where: [via Matrix](https://matrix.to/#/#project-mesa:matrix.org)

Proposed Agenda:

-Code of Conduct/Governance
-Small Development Grant and Visualization Improvements
-Mesa-Geo and CSSSA
-PRs and Issues

@rht , @jackiekazil , @Corvince , @wang-boyu , @ankitk50, @rlskoeser

**Comment by wang-boyu** [comment]
Adding @SongshGeo who may also want to join us

**Reply by SongshGeo** [reply]
Thanks for your notification. I'd like to be in.

**Reply by tpike3** [reply]
@SongshGeo Welcome, looking forward to meeting you; Thanks @wang-boyu

**Comment by tpike3** [comment]
Code of Conduct References: 

Overview: https://opensource.guide/code-of-conduct/ 

Staying positive: https://github.com/jonschlinkert/maintainers-guide-to-staying-positive 

Examples: 
- https://opensource.com/code-of-conduct 
- https://www.contributor-covenant.org/version/2/1/code_of_conduct/
- https://www.apache.org/foundation/policies/conduct
- https://aws.github.io/code-of-conduct

**Reply by jackiekazil** [reply]
The requirement for whatever direction we take is to ensure we have an explicit plan for enforcement, otherwise the CoC is worthless.

**Reply by wang-boyu** [reply]
Some references on open-source project governance (I'm interested in how people are organized around a project and how code contributions are reviewed and merged):

- [Governance models for open source projects](https://github.com/community/maintainers/discussions/121) (may need to join the Maintainer Community on GitHub to access it)
- [Healthy Open Source](https://medium.com/the-node-js-collection/healthy-open-source-967fa8be7951): A walkthrough of the Node.js Foundation’s base contribution policy
- [Apache Forrest project guidelines](https://forrest.apache.org/guidelines.html)
- [Governance Models](http://oss-watch.ac.uk/resources/governancemodels) from OSS Watch

**Reply by tpike3** [reply]
Sorry for the delay on this. We do have a [Code of Conduct](https://github.com/projectmesa/mesa/blob/main/CODE_OF_CONDUCT.md), but I did make another variation. More importantly and to @wang-boyu  we needed a governance document. 

Appreciating people won't have time to read this, I would recommend looking at lines 54 -59 and 77-80. Or more broadly the contributor and committer descriptions. The critical point is how do we in place mechanisms that address concerns that have occurred over many years about the timeliness of our merges and the ease of contribution. One element that needs to be added is the expectation of response to a contribution-- 1 week?   

[Mesa Governance (Draft).md](https://github.com/projectmesa/mesa/files/12903840/Mesa.Governance.Draft.md)
[Mesa Code of Conduct (Draft).md](https://github.com/projectmesa/mesa/files/12903841/Mesa.Code.of.Conduct.Draft.md)


Let me know what you think.

**Reply by rht** [reply]
Regarding with the code of conduct, it's more relevant to use [NUMFOCUS](https://numfocus.org/code-of-conduct)'s instead of PSF's.
> We require all NumFOCUS sponsored and affiliated projects to adopt a code of conduct for their project that encourages a productive, respectful environment for all open source contributors and community members.

>  One element that needs to be added is the expectation of response to a contribution-- 1 week?

This is not an issue. I usually responded to most PR's and issues' within reasonable time, often much less than a week.

**Reply by jackiekazil** [reply]
All -- I apologize that I was not in the meeting yesterday - I was traveling and it fell off my radar. 

@tpike3 thank you for posting these. Can you put them in a branch or PR for easier review and comment?

**Reply by tpike3** [reply]
@jackiekazil -- done to keep it focused I just did the governance document in a pull request https://github.com/projectmesa/mesa/pull/1836
